src/ed25519.rs

```diff
@@ -447,6 +447,12 @@ impl Keypair {
     }
 }
 
+impl PartialEq for Keypair {
+    fn eq(&self, other: &Self) -> bool {
+        self.public.as_ref() == other.public.as_ref()
```

Comment:
I have verified that I can compare Keypairs with this change from an external crate.

Reply:
Not sure how much this actually matters in practice, but due to clamping differing canonical private scalars may result in the same public key.

Since `public` is a `pub` field anyway, is there any reason why you can't just compare public keys if what you care about is public key equality?

Reply:
thanks for the feedback.

seems to me that public key collisions wouldn't be a concern in any arena where these keys are  in use for real work.

that said, comparison's are only being done in tests, generally, where assert_eq!() wants to see ParialEq, which I can't implement for an external crate

Reply:
Why can't you `assert_eq!(&keypair1.public, &keypair2.public)`?

It seems like what you care about is public key equality (at least per this implementation), so given that, the above implementation seems to better reflect the intent.

Reply:
ah: because I'm not comparing keypairs directly, I have a keypair in a lot of otherwise PartialEq structs...